### DISCUSSION TITLE
How to make sure a resource is unique per user (address)?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#172] [Questions] [answered]
### Describe your question in detail.

Let's say I'm building a social app, I have a `User` struct that is unique per user. I want to store it under an object owned by user instead of directly under user. 

Is named object the only option?  Because named object is the only one that generate address based on seeds so I can set seeds to be `[contract_addr, user_addr, "user"]`. Then in `sign_up` function the named object doesn't exist means user hasn't signed up before. 

If I don't use named object, I think we have to use some user registry which is not practical because it's too slow?

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

N/A

**Comment by kshitijc1506** [comment]
With the current state of things, would this be feasible because the object can just be transferred right? Is it not better to use soulbound tokens here?

**Reply by 0x-j** [reply]
Similar to token, object also has a transfer ref you can disable transfer. I'm more wondering can we ensure uniqueness without named using named object.

**Reply by gregnazario** [reply, marked as answer]
You can only find uniqueness by using either:
1. A named object (soulbound)
2. A resource that holds the object to the object (and therefore is unique on the resource)

These both essentially use a consistent hash to determine whether you already have the item
